cli/module_loader.rs

```diff
@@ -118,7 +124,14 @@ impl ModuleLoader for CliModuleLoader {
     let state = op_state.borrow();
 
     // The permissions that should be applied to any dynamically imported module
-    let dynamic_permissions = state.borrow::<Permissions>().clone();
+    let dynamic_permissions =
+      if let Some(permissions) = self.permissions.as_ref().take() {
```

Comment:
`.as_ref()` is superfluous

Reply:
How can I replace it? Borrowing self as mut requires me to change the original ModuleLoader implementation as well

Reply:
You're right. Then it suggest that current version also doesn't work as expected...

Reply:
Is there a test that checks if dynamic imports in worker use worker's permission instead of parents?

Reply:
No, it only checks if it has read permission with `Deno.permissions.query`. If it's a separate matter, I'll add a test